Colin

coldog86 opened the pull request:
Unit tests and integration test for entry are done. Entry with a season ticket is not part of this.

coldog86 committed:
Create buttonPushedTest_NotWorking

coldog86 committed:
Update buttonPushedTest_NotWorking

coldog86 committed:
Update buttonPushedTest_NotWorking

coldog86 committed:
Update EntryController.java

coldog86 committed:
Create testTicketTaken

coldog86 committed:
Create testCarEventDetected

coldog86 committed:
Create testEntryControllerWithNulls

coldog86 committed:
Create testEntryControllerStatesWorkAsExpected

coldog86 committed:
Delete buttonPushedTest_NotWorking

coldog86 committed:
Delete testCarEventDetected

coldog86 committed:
Delete testEntryControllerStatesWorkAsExpected

coldog86 committed:
Delete testEntryControllerWithNulls

coldog86 committed:
Delete testTicketTaken

coldog86 committed:
added a getter for number of cars parked

coldog86 committed:
Added a getter for number of cars parked

coldog86 committed:
added getter for state

coldog86 committed:
add setter for carDetected boolean

coldog86 committed:
AdhocTicket unit tests

coldog86 committed:
unit tests for buttonPushed()

test that the button checks carpark is at capacity first
test When carpark is full it will display correct message and change state
test When not full it will issue a ticket, display the correct message and change state
test that deviation to norm will cause beep()

coldog86 committed:
tests for the entry controller

test state changes are functioning
tests constructor with and without nulls

coldog86 committed:
tests the ticketTaken method

coldog86 committed:
Intergration tests for entry use cases

Tests the Normal flow
tests Alternate Flow where the Carpark is full and the customer waits for someone to leave
tests Alternate Flow where the Carpark is full and the customer leaves
tests Alternate Flow where the customer drives up to the gate, then backs out
tests Alternate Flow where the customer drives up to the gate, pushes the button, then backs out

Abrody01 commented on src/AdhocTicketTests:
```diff
@@ -0,0 +1,106 @@
+            mockOS = mock(CarSensor.class);
+            mockUI = mock(IEntryUI.class);
+            mockTicket = mock(IAdhocTicket.class);
+            mockEntry = new EntryController(mockCarpark, mockGate, mockOS, mockIS, mockUI);
```
Recommended to change variable name from mockEntry to testEntry, This way it is clearer what is being mocked vs tested.

Abrody01 commented on src/AdhocTicketTests:
```diff
@@ -0,0 +1,106 @@
+import bcccp.carpark.CarSensor;
```
I don't think you need to have this test at all.

Abrody01 commented on src/EntryControllerTest:
```diff
@@ -0,0 +1,337 @@
+            CarSensor mockOS;
+            IEntryUI mockUI;
+            IAdhocTicket mockTicket;
+            EntryController mockEntry;
```
Recommend changing "mockEntry" to "testEntry" so it is more clear what you are testing

coldog86 replied:
resolved

Abrody01 commented on src/EntryControllerTest:
```diff
@@ -0,0 +1,337 @@
+        
+    }
+    
+    
```
I would recommend splitting this file out to a few components, one for 'state transition' and one for the constructor.

coldog86 replied:
resolved

Abrody01 reviewed (changes requested):
I think you should also the user entering a season ticket along with the button pushed test's. This is in the use case as part 3.2 and 7.2

coldog86 committed:
Update EntryIntergrationTest

coldog86 committed:
removed, to be split up and added back in

coldog86 committed:
Create EntryControllerConstructorTest

Entry Controller tests for the constructor

coldog86 committed:
EntryController Transition Tests

coldog86 committed:
User acceptance tests

season ticket entry, normal flow
adhock ticket entry, normal flow
adhock ticket entry, carpark full, customer waits

Abrody01 reviewed (approved):
All good to go now great work Colin
